shopfloor/services/zone_picking.py

```diff
@@ -769,9 +769,15 @@ def set_destination(
                 pkg_moved, response = self._set_destination_location(
                     move_line, quantity, confirmation, location,
                 )
-                if response:
-                    if extra_message:
+                if response and extra_message:
+                    if response.get("message"):
                         response["message"]["body"] += "\n" + extra_message["body"]
+                    else:
+                        # _handle_pick_pack_same_time_for_location successful
+                        response["message"] = {
+                            "body": extra_message["body"],
+                            "message_type": "success",
+                        }
```

Comment:
if there no `response['message']` just do `response['message'] = extra_message`, you should not enforce the message type here.